app/models/sample.rb

```diff
@@ -747,7 +748,13 @@ def kickoff_pipeline
 
     pr.alignment_config = AlignmentConfig.find_by(name: alignment_config_name) if alignment_config_name
     pr.alignment_config ||= AlignmentConfig.find_by(name: AlignmentConfig::DEFAULT_NAME)
-    pr.save
+    pr.save!
+  rescue StandardError => err
+    LogUtil.log_err_and_airbrake("Error saving pipeline run: #{err.inspect}")
+    # This will cause a message to be shown to the user on the sample page.
+    # See app/assets/src/components/utils/sample.js
+    self.upload_error = Sample::UPLOAD_ERROR_PIPELINE_KICKOFF
+    save!
```

Comment:
This code is being run on a `before_save` callback and you call save again here. Seems like a dangerous pattern. What is the status of the sample at this point?

Reply:
Good catch! I will remove the save. It should still work.